lib/jerboa/format/body/attribute.ex

```diff
@@ -3,14 +3,31 @@ defmodule Jerboa.Format.Body.Attribute do
   STUN protocol attributes
   """
 
-  alias Jerboa.Format.Body.Attribute
   alias Jerboa.Format.ComprehensionError
   alias Jerboa.Params
 
+  defprotocol EncoderProtocol do
+    @spec type_code(t) :: integer
+    def type_code(attr)
+
+    @spec encode(t, Params.t) :: binary
+    def encode(attr, params)
+  end
+
+  defprotocol DecoderProtocol do
+    @spec decode(type :: t, value :: binary, params :: Params.t)
+    :: {:ok, t} | {:error, struct} | :ignore
+    def decode(type, value, params)
+  end
+
+  @apps_lib_dirs  [:code.lib_dir(:jerboa, :ebin)]
```

Comment:
Look at my last comment. The code is not complete. We should somehow allow a user to provide his apps, maybe via `use`.